lib/ansible/modules/cloud/docker/docker_plugin.py

```diff
@@ -0,0 +1,297 @@
+
+        self.dclient = DockerClient()
+        self.dclient.api = client
+
+        self.parameters = TaskParameters(client)
+        self.check_mode = self.client.check_mode
+        self.results = {
+            u'changed': False,
+            u'actions': []
+        }
+        self.diff = self.client.module._diff
+
+        self.existing_plugin = self.get_existing_plugin()
+
+        state = self.parameters.state
+        if state == 'present':
+            self.present()
+        elif state == 'absent':
+            self.absent()
+        elif state == 'enable':
+            self.enable()
+        elif state == 'disable':
+            self.disable()
+
+    def get_plugin_name(self):
+        return self.parameters.alias or self.parameters.name
+
+    def get_existing_plugin(self):
+        name = self.get_plugin_name()
+        try:
+            plugin = self.dclient.plugins.get(name)
+        except NotFound:
+            return None
+        except APIError as e:
+            self.client.fail(text_type(e))
+
+        return plugin
+
+    def has_different_config(self):
+        """
+        Return the list of differences between the current parameters and the existing volume.
+
+        :return: list of options that differ
+        """
+        differences = []
+        if self.parameters.plugin_options:
+            if not self.existing_plugin.settings:
+                differences.append('plugin_options')
+            else:
+                existing_options_list = self.existing_plugin.settings['Env']
+                existing_options = parse_options(existing_options_list)
+
+                for key, value in iteritems(self.parameters.plugin_options):
+                    if ((not existing_options.get(key) and value) or
+                            not value or
+                            value != existing_options[key]):
+                        differences.append('plugin_settings.%s' % key)
+
+        return differences
+
+    def install_plugin(self):
+        if not self.existing_plugin:
+            if not self.check_mode:
+                try:
+                    self.existing_plugin = self.dclient.plugins.install(self.parameters.plugin_name, self.parameters.plugin_alias)
+                except APIError as e:
+                    self.client.fail(text_type(e))
+
+            self.results['actions'].append("Installed plugin %s" % self.get_plugin_name())
+            self.results['changed'] = True
+
+    def remove_plugin(self):
+        if self.existing_plugin:
+            if not self.check_mode:
+                try:
+                    self.existing_plugin.remove()
+                except APIError as e:
+                    self.client.fail(text_type(e))
+
+            self.results['actions'].append("Removed plugin %s" % self.get_plugin_name())
+            self.results['changed'] = True
+
+    def update_plugin(self):
+        if not self.check_mode:
+            try:
+                if self.existing_plugin.enabled:
```

Comment:
You should check if the options actually changed before reconfiguring the plugin. It only makes sense to reconfigure if this actually changes something.